homeassistant/components/shell_command.py

```diff
@@ -64,6 +62,6 @@ def _parse_command(hass, cmd, variables):
         shell = True
     else:
         # template used. Must break into list and use shell=False for security
-        cmd = [prog] + rendered_args.split()
+        cmd = [prog] + shlex.split(rendered_args)
```

Comment:
Using spaces results in more arguments than expected:

Using service data on the dev console: 

service data (dev console):
```json
{"message": "does it work"}
```

configuration.yaml:
```yaml
shell_command:
  data_test: '/home/pi/test.js "testing" message="{{ message }}"'
#!/usr/bin/env node
```

test.js:
```javascript
var fs = require('fs');

fs.writeFileSync('/home/pi/test.txt', 'Arguments:\n\n' + process.argv.join('\n') + '\n', 'utf8');
```

result test.txt:
```
Arguments:

/usr/local/bin/node
/home/pi/test.js
"testing"
message="does
it
work"
```